### PR TITLE
Implement field attributes

### LINKED ISSUE #11: Expose access to attributes on reflect::Field
For example I should be able to write the following to look for `#[serde(skip_serializing)]`:

```rust
for field in receiver.fields() {
    if has_skip_serializing(&field) {
        continue;
    }
    /* serialize the field */
}

fn has_skip_serializing(field: &reflect::Field) -> bool {
    ???
}
```

### PR DESCRIPTION
**Pull request by eupn**
Closes #11.

### CODE REVIEW

File: src/attr.rs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        args: Vec<String>,
+    },
+
+    /// Attributes in other forms which aren't calls
+    Other {
+        tokens: String,
+    },
+}
+
+pub(crate) fn convert_attributes(attrs: &[syn::Attribute]) -> Vec<Attribute> {
```
Comment (dtolnay) [inline src/attr.rs]: How about keeping them as syn::Attribute? I don't see a reason to do this conversion.

**Comment by dtolnay** [comment]
Thanks, this looks along the lines of what I had in mind except for the type they are exposed as.

File: src/attr.rs (hunk 1 above)
Comment (eupn) [reply src/attr.rs]: It didn't seem to me as straightforward as I wanted it to be to work with the field attributes from the perspective of a macro author so I decided to abstract implementation details (`syn::Attribute`) away with this.

Comment (8BitMate) [reply src/attr.rs]: I don't have strong opinions either way. I see eupn's point of making it easier for the macro author for the simple case, but syn::Attribute offers a bit more flexibility. I get that Other is a catch all and can be parsed further, but with syn::Attribute, there is already some added structure.

Comment (eupn) [reply src/attr.rs]: Got you guys, I'll change that.

File: tests/test_field_access.rs [hunk 2]
```diff
@@ -1,14 +1,47 @@
...
+                                        .map(|i| i.to_token_stream().to_string())
+                                        .find(|tok| tok == "skip")
+                                        .is_some()
+                            }
+                            _ => false,
+                        };
+
+                        Ok(res)
+                    })
+                    .unwrap_or(false)
```
Comment (dtolnay) [inline tests/test_field_access.rs]: I would recommend writing this as:

```rust
syn::custom_keyword!(skip);
attr.path.is_ident("trivial") && attr.parse_args::<skip>().is_ok()
```

**Pull request merged**